src/ScottPlot/Plottable/SignalPlotList.cs

```diff
@@ -0,0 +1,129 @@
+﻿using ScottPlot.MinMaxSearchStrategies;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ScottPlot.Plottable
+{
+    /// <summary>
+    /// A SignalPlotList is a SignalPlot that is designed to grow using Add() and AddRange() methods
+    /// </summary>
+    public class SignalPlotList : SignalPlotBase<double>
+    {
+        /// <summary>
+        /// The total number of values the list can hold without resizing.
+        /// </summary>
+        public int Capacity => Ys.Length;
+
+        /// <summary>
+        /// Number of values in the list.
+        /// This number will never exceed the Capacity.
+        /// </summary>
+        public int Count => NextPointIndex;
+
+        /// <summary>
+        /// Percentage of the capacity currently filled with data
+        /// </summary>
+        public double FillPercentage => 100.0 * Count / Capacity;
+
+        /// <summary>
+        /// Return the last value in the list (or null if empty)
+        /// </summary>
+        public double? LastY => (Count > 0) ? Ys[NextPointIndex - 1] : null;
+
+        /// <summary>
+        /// Return the time point fot the last value in the list (or null if empty)
+        /// </summary>
+        public double? LastX => (Count > 0) ? (NextPointIndex - 1) / SampleRate + OffsetX : null;
+
+        /// <summary>
+        /// Index in the Ys array where the next data point will be placed.
+        /// </summary>
+        private int NextPointIndex = 0;
+
+        /// <summary>
+        /// When the Ys array must be expanded, increase its length by this number of values.
+        /// </summary>
+        private readonly int CapacityIncriment;
```

Comment:
# Closed Form Representation

A closed form representation of the number of resizes for each function is below:
Constant Increment: 
![image](https://user-images.githubusercontent.com/8635304/108643169-89a96b80-7466-11eb-8f68-bc607baf1626.png)

Constant Multiple:
![image](https://user-images.githubusercontent.com/8635304/108643289-37b51580-7467-11eb-9cb3-11ca93ab1420.png)

# Cost of Pushing

Simply multiply the cost of resizing by the frequency of resizing:

![image](https://user-images.githubusercontent.com/8635304/108643258-0d635800-7467-11eb-9cd6-4cbbd81a7567.png)
![image](https://user-images.githubusercontent.com/8635304/108643283-2d931700-7467-11eb-9f4b-3c7a68c366ca.png)

Reply:
You can change the resizing algorithm, but it'd likely be easier for you and the user to just allow them to pass in a `List<double>`, and then you don't need to manage any memory yourself. Plus the BCL can probably be faster because it'll be using unmanaged buffers, not C# arrays.

Reply:
Hey @bclehmann, 

These are excellent points! I considered the growth rate, but went with the linear model here because it seemed logical considering that Signal plot data is expected to come in at a fixed rate (`SampleRate`), so being able to set that would let the user intuit how often the array will be copied. A counter argument to the exponential growth model is that it's wasteful when the arrays get large. I guess the fundamental trade off is CPU vs. memory? Considering how slow plotting is, an occasional array copy doesn't seem likely to make a big difference either way, so it may make sense to favor conserving memory here. Ultimately it probably doesn't matter much one way or another, but this is very good to consider! 

Maybe an intermediate option is a hybrid solution that uses two public fields?
```cs
public float GrowthFactor = 1.5;
public int MaximumCapacityIncrement = 1e6;
```

Your last comment noted "Allow them to pass in `List<double>`", which would be great, but the signal plots use min/max algorithms that require double or generic arrays, making it difficult to support lists #526, #447

Reply:
You cannot use `<`  or `>` with generics, however, you _can_ define a comparison function. It would be a pain for the user to have to specify a comparison function, but I suppose you could have a dictionary of common ones (i.e. numeric types).

Also, I'm not sure how this applies, I was suggesting `List<double>`, not `List<T>`. You have the same problem with comparison operators with `T[]` as you do with `List<T>`, even if we did want to allow users to pass in `int` or `float` (or whatever new floating point format NVidia comes up with next). Perhaps I misunderstood what you're saying?

Reply:
Hi @bclehmann,
> You cannot use < or > with generics, however, you can define a comparator function. It would be a pain for the user to have to specify a comparator function, but I suppose you could have a dictionary of common ones (i.e. numeric types).

>Also, I'm not sure how this applies, I was suggesting List<double>, not List<T>. You have the same problem with comparison operators with T[] as you do with List<T>, even if we did want to allow users to pass in int or float (or whatever new floating point format NVidia comes up with next). Perhaps I misunderstood what you're saying?

This problem is solvable, and has already been solved for `Signal<T>`.

>One hack from Jon Skeet is to use compiled at runtime lambda expressions. At runtime type of T already can be taken, and you can manualy compile any math on it to high performance IL code and call it. Thats make code complex and unreadable, and compiler cant check for errors. But using lambda reduce performance drop to only 10-30% compare to nongeneric implementation.